### ISSUE TITLE
UI: Scale vm failed if the first disk offering is customized

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#8212]
Steps to reproduce the issue
- remove built-int disk offerings (Small, Medium, Large)
- create a custom disk offering if the 'Custom' offering does not exist
- create a vm
- stop the vm, and resize the root disk size
- scale the vm to another service offering, the disk size is displayed incorrectly
![image](https://github.com/apache/cloudstack/assets/57355700/d29d3082-90e3-4e2c-a547-45ed8c48ca32)

- click OK, got an error "You cannot pass in a custom disk size to a non-custom disk offering."
![image](https://github.com/apache/cloudstack/assets/57355700/383cad3b-7240-4bac-89b7-c220c906c9a4)

In 4.18 and onwards, update global setting allow.diskoffering.change.during.scale.vm to true (this does not exist in 4.17) before testing.

logs show that the `details[0].rootdisksize=0` is passed which should not be passed.

```
2023-11-10 07:54:46,650 DEBUG [c.c.a.ApiServlet] (qtp989447607-16:ctx-b67707d1) (logid:13ab8e58) ===START===  127.0.0.1 -- GET  id=ab516f09-de84-407d-8a15-a4685c5d17f1&serviceofferingid=60f2bf60-30b6-4117-a265-b1b1b7f00a94&automigrate=true&details[0].rootdisksize=0&command=scaleVirtualMachine&response=json
```

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
4.17/4.18/main
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
cloudmonkey commands
```
(localcloud) 🐱 > update configuration name=allow.diskoffering.change.during.scale.vm value=true
{
  "configuration": {
    "category": "Advanced",
    "component": "UserVmManager",
    "defaultvalue": "false",
    "description": "Determines whether to allow or disallow disk offering change for root volume during scaling of a stopped or running vm",
    "displaytext": "Allow diskoffering change during scale vm",
    "group": "Storage",
    "isdynamic": true,
    "name": "allow.diskoffering.change.during.scale.vm",
    "subgroup": "Volume",
    "type": "Boolean",
    "value": "true"
  }
}



(localcloud) 🐱 > scale virtualmachine id=eaf2c009-3286-4e88-87b2-1e78bc7d124c details[0].rootdisksize=0 serviceofferingid=4686773d-b8c6-47ba-9a3b-591c2c9aa860 
{
  "account": "admin",
  "accountid": "19d46742-417f-11ee-bf06-1e007c0000fc",
  "cmd": "org.apache.cloudstack.api.command.admin.vm.ScaleVMCmdByAdmin",
  "completed": "2023-11-10T08:03:45+0000",
  "created": "2023-11-10T08:03:45+0000",
  "domainid": "f1f66ec7-417e-11ee-bf06-1e007c0000fc",
  "domainpath": "ROOT",
  "jobid": "b830f90e-8f44-4047-8890-0dd1058ab6c5",
  "jobinstanceid": "eaf2c009-3286-4e88-87b2-1e78bc7d124c",
  "jobinstancetype": "VirtualMachine",
  "jobprocstatus": 0,
  "jobresult": {
    "errorcode": 431,
    "errortext": "You cannot pass in a custom disk size to a non-custom disk offering."
  },
  "jobresultcode": 431,
  "jobresulttype": "object",
  "jobstatus": 2,
  "userid": "19d55e1b-417f-11ee-bf06-1e007c0000fc"
}
🙈 Error: async API failed for job b830f90e-8f44-4047-8890-0dd1058ab6c5

```

**Comment by DaanHoogland** [comment]
fixed in #8213
